curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-pyramid-generator/660f1cedf3676fe26122ebf6.md

```diff
@@ -0,0 +1,54 @@
+---
+id: 660f1cedf3676fe26122ebf6
+title: Step 33
+challengeType: 0
+dashedName: step-33
+---
+
+# --description--
+
+You should see the numbers zero through seven printed in your console, one per line. This will serve as the foundation for generating your pyramid.
+
+Replace your log statement with a statement to push `i` to your `rows` array.
+
+# --hints--
+
+You should not have a `console.log` call.
+
+```js
+assert.notMatch(code, /console/);
+```
+
+You should call `.push()` on your `rows` array.
+
+```js
+assert.match(code, /rows\.push\(/);
+```
+
+You should push `i` to your `rows` array.
+
+```js
+assert.match(code, /rows\.push\(\s*i\s*\)/);
+```
+
+Your `.push()` should happen in your `for` loop. Don't forget your semi-colon.
+
+```js
+assert.match(code, /for\s*\(\s*let\s+i\s*=\s*0;\s*i\s*<\s*count;\s*i\s*=\s*i\s*\+\s*1\s*\)\s*\{\s*rows\.push\(\s*i\s*\);\s*\}/)
+```
```

Comment:
````suggestion
Your `.push()` should happen in your `for` loop. 

```js
assert.match(code, /for\s*\(\s*let\s+i\s*=\s*0;\s*i\s*<\s*count;\s*i\s*=\s*i\s*\+\s*1\s*\)\s*\{\s*rows\.push\(\s*i\s*\);?\s*\}/)
```
````